File: test_payload.cs
using System;
using System.IO;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

class Program
{
    static async Task Main()
    {
        var key = "6Ws98iJ3hijXQ5OtcZbIEJLKklcqZMJ2G1iIjTBSVY73aI3GRyElJQQJ99CBACHYHv6XJ3w3AAAAACOGPSvn"; // appsettings.json Key
        var region = "eastus2";
        var url = $"https://{region}.api.cognitive.microsoft.com/speechtotext/transcriptions:transcribe?api-version=2025-10-15";

        var audioBytes = new byte[8000]; // Dummy silent buffer
        for (int i = 0; i < audioBytes.Length; i++) audioBytes[i] = 128; // PCM silent

        using var client = new HttpClient();
        using var content = new MultipartFormDataContent();
        
        var audioContent = new ByteArrayContent(audioBytes);
        audioContent.Headers.ContentType = MediaTypeHeaderValue.Parse("audio/wav");
        content.Add(audioContent, "audio", "audio.wav");

        var definitionJson = "{\"locales\":[\"pt-BR\"],\"profanityFilterMode\":\"None\",\"diarization\":{\"enabled\":true,\"maxSpeakers\":2}}";
        content.Add(new StringContent(definitionJson, Encoding.UTF8, "application/json"), "definition");

        using var request = new HttpRequestMessage(HttpMethod.Post, url);
        request.Headers.Add("Ocp-Apim-Subscription-Key", key);
        request.Content = content;

        Console.WriteLine("Enviando requisicao...");
        var response = await client.SendAsync(request);
        var json = await response.Content.ReadAsStringAsync();
        Console.WriteLine(json);
    }
}

File: test_payload.py
import urllib.request
import json
import uuid
import sys
import os

url = "https://eastus2.api.cognitive.microsoft.com/speechtotext/transcriptions:transcribe?api-version=2025-10-15"
key = "6Ws98iJ3hijXQ5OtcZbIEJLKklcqZMJ2G1iIjTBSVY73aI3GRyElJQQJ99CBACHYHv6XJ3w3AAAAACOGPSvn"
boundary = uuid.uuid4().hex
audio_path = r"Backend\wwwroot\uploads\audio\11d88381-9859-44f3-b33d-10cde7809cce_audio_completo_merged.wav"

if not os.path.exists(audio_path):
    print(f"File {audio_path} not found!")
    sys.exit(1)

with open(audio_path, "rb") as f:
    wav_data = f.read()[:5000000] # Take first 5MB just to be safe and fast

definition = '{"locales":["pt-BR"],"profanityFilterMode":"None","diarization":{"enabled":true,"maxSpeakers":2}}'

body = (
    f"--{boundary}\r\n"
    f'Content-Disposition: form-data; name="audio"; filename="audio.wav"\r\n'
    f"Content-Type: audio/wav\r\n\r\n"
).encode('utf-8') + wav_data + (
    f"\r\n--{boundary}\r\n"
    f'Content-Disposition: form-data; name="definition"\r\n'
    f"Content-Type: application/json\r\n\r\n"
    f"{definition}\r\n"
    f"--{boundary}--\r\n"
).encode('utf-8')

req = urllib.request.Request(url, data=body, headers={
    'Ocp-Apim-Subscription-Key': key,
    'Content-Type': f'multipart/form-data; boundary={boundary}'
}, method='POST')

try:
    with urllib.request.urlopen(req) as response:
        with open("payload_output.json", "w", encoding='utf-8') as out:
            out.write(response.read().decode('utf-8'))
        print("Success")
except urllib.error.HTTPError as e:
    print(e.read().decode('utf-8'))
